timemachine/cpp/src/tibd_exchange_move.cu

```diff
@@ -48,10 +48,13 @@ TIBDExchangeMove<RealType>::TIBDExchangeMove(
       radius_(static_cast<RealType>(radius)), inner_volume_(static_cast<RealType>((4.0 / 3.0) * M_PI * pow(radius, 3))),
       d_rand_states_(DEFAULT_THREADS_PER_BLOCK), d_inner_mols_count_(1), d_identify_indices_(this->num_target_mols_),
       d_partitioned_indices_(this->num_target_mols_), d_temp_storage_buffer_(0), d_center_(3),
-      d_uniform_noise_buffer_(round_up_even(NOISE_PER_STEP * this->proposals_per_move_)), d_targeting_inner_vol_(1),
-      d_ligand_idxs_(ligand_idxs), d_src_weights_(this->num_target_mols_), d_dest_weights_(this->num_target_mols_),
+      d_uniform_noise_buffer_(round_up_even(NOISE_PER_STEP * this->proposals_per_move_ * this->samples_per_proposal_)),
+      d_targeting_inner_vol_(this->samples_per_proposal_), d_ligand_idxs_(ligand_idxs),
+      d_src_weights_(this->num_target_mols_), d_dest_weights_(this->num_target_mols_),
       d_inner_flags_(this->num_target_mols_), d_box_volume_(1), p_inner_count_(1), p_targeting_inner_vol_(1),
-      d_selected_translation_(3) {
+      d_selected_translation_(this->samples_per_proposal_ * 3),
+      d_sample_after_segments_(this->d_sample_segments_.length), d_weights_before_counts_(this->samples_per_proposal_),
+      d_weights_after_counts_(this->samples_per_proposal_) {
```

Comment:
`d_weights_after_counts_` is currently unused, but is set up, as I got ahead of myself with these changes. I left it in as currently only takes up one float/double and seemed sensible to set up at the same time as `d_weights_before_counts_`. Happy to remove if it seems confusing